src/asm_pattern.py

```diff
@@ -193,15 +193,30 @@ def derived_meta(self) -> InstructionMeta:
         return InstructionMeta.missing()
 
     def apply(self, repl: Replacement, arch: ArchAsm) -> None:
-        # TODO: It is possible to check the output registers of the replaced input
-        # instructions to determine a set of clobbered registers.
+        # Track which registers are overwritten/clobbered in the replacement asm
+        repl_writes = []
+        final_instr: Optional[Instruction] = None
         for part in repl.new_body:
             if isinstance(part, AsmInstruction):
                 # Parse any AsmInstructions into Instructions before substituting
-                instr = arch.parse(part.mnemonic, part.args, self.derived_meta())
-                self.output.append(instr)
-            else:
-                self.output.append(part)
+                part = arch.parse(part.mnemonic, part.args, self.derived_meta())
+            if isinstance(part, Instruction):
+                # Update list of written registers
+                repl_writes.extend(part.outputs)
+                repl_writes.extend(part.clobbers)
+                final_instr = part
+            self.output.append(part)
+
+        # Calculate which regs are *not* written by the repl asm, but were in the input asm
+        # Denote the replacement asm as "clobbering" these regs by marking the final instr
```

Comment:
I should add, nice touch! I really like the extra error prevention here.

Reply:
I meant to post it before, but this change was responsible for exactly 1 diff in MM/PM, [in PM's `collision_main_lateral`](https://gist.github.com/zbanks/fdb4fe2217a9f1416e630f8648c694c0)